app/iot/service.py

```diff
@@ -18,32 +18,54 @@ def connect(self) -> None:
     def disconnect(self) -> None:
         ...
 
-    def send_message(self, message_type: MessageType, data: str) -> None:
+    async def send_message(self, message_type: MessageType, data: str) -> None:
         ...
 
 
 class IOTService:
     def __init__(self) -> None:
         self.devices: dict[str, Device] = {}
 
-    def register_device(self, device: Device) -> str:
-        device.connect()
+    async def register_device(self, device: Device) -> str:
+        await device.connect()
         device_id = generate_id()
+        attempts_left = 7
+        while device_id in self.devices:
```

Comment:
Bug: This loop can potentially become infinite if the generated device_id is always in self.devices. Consider adding a limit to the number of attempts or a different strategy for generating a unique device_id.